core/src/components/list/list.tsx

```diff
@@ -43,9 +54,27 @@ export class List implements ComponentInterface {
     return false;
   }
 
+  private getShape(): string | undefined {
+    const theme = getIonTheme(this);
+    const { shape } = this;
+
+    // TODO(ROU-10831): Remove theme check when shapes are defined for all themes.
+    if (theme !== 'ionic') {
+      return undefined;
+    }
+
+    if (shape === undefined) {
+      return 'round';
```

Comment:
The task's AC says that:

> The default value changes based on the property inset
> inset=false (default) → default value: Rectangular
> inset=true → default value: Round

Shouldn't this be tackled in this method?

Reply:
Good catch! https://github.com/ionic-team/ionic-framework/pull/29622/commits/ede0fcecc189ce74cbd8d2a2597131607d53f1bc

Reply:
@OS-giulianasilva ready for review